jpa/eclipselink/src/main/java/example/springdata/jpa/eclipselink/domain/Participant.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package example.springdata.jpa.eclipselink.domain;
+
+/**
+ * Entity showing use of {@link org.eclipse.persistence.annotations.UuidGenerator} from
+ * the EclipseLink JPA Provider. Does not extend the AbstractPersistable because of the UUID generator
+ *
+ * @author Jeremy Rickard
+ */
+
+import org.eclipse.persistence.annotations.UuidGenerator;
+
+import javax.persistence.*;
+import java.util.List;
+
+@Entity
+public class Participant {
+    @Id
+    @UuidGenerator(name = "UUID")
+    @GeneratedValue(generator = "UUID")
+    private String uuid;
+
+    private String firstName;
+
+    private String lastName;
+
+    private String emailAddress;
+
+    private String phoneNumber;
+
+    private int age;
+
+    @Enumerated
+    private Gender gender;
+
+    @ManyToMany(mappedBy = "registrations")
+    private List<Race> raceRegistrations;
+
+    public String getUuid() {
```

Comment:
Let's use Lombok annotations to avoid all the boilerplate getters and setters. Might be work having a look how other JPA samples do this.